deye_daemon.py

```diff
@@ -90,12 +93,35 @@ def __remove_duplicated_reg_ranges(self, reg_ranges: list[SensorRegisterRange])
         return result
 
 
+class IntervalRunner:
+    def __init__(self, interval, action):
+        self.__log = logging.getLogger(DeyeDaemon.__name__)
+        self.interval = interval
+        self.action = action
+        self.stopEvent = threading.Event()
+        thread = threading.Thread(target=self.__setInterval)
+        self.__log.debug("Start to execute the daemon at intervals of %s seconds", self.interval)
+        thread.start()
+
+    def __setInterval(self):
+        nextTime = time.time() + self.interval
+        while not self.stopEvent.wait(nextTime - time.time()):
+            nextTime += self.interval
+            try:
+                self.action()
+            except Exception as e:
+                self.__log.exception("Unexpected error during daemon execution")
```

Comment:
@palto42 Could you also pass `e` to the exception method? This way it will be more visible where it crashed.

Reply:
@kbialek It is actually not required to pass `e` in `logging.exception`, the logger automatically adds it to the message, see [loggin.exception](https://docs.python.org/3.10/library/logging.html#logging.exception).

Actually I found that the way you use it in `deye_connector.py` is incorrect and causes an exception in the logger:

Example:

```python
❯ python
Python 3.10.6 (main, Mar 10 2023, 10:55:28) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import logging
>>> try:
...     1/0
... except Exception:
...     logging.exception("Something unexpected happened")
... 
ERROR:root:Something unexpected happened
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
ZeroDivisionError: division by zero
>>> 
>>>
>>> try:
...     1/0
... except Exception as e:
...     logging.exception("Something unexpected happened: %s", e)
... 
ERROR:root:Something unexpected happened: division by zero
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
ZeroDivisionError: division by zero
>>>
```
Actually I found that the way you use it in `deye_connector.py` is wrong and causes an exception in the logger:

```python
>>> try:
...     1/0
... except Exception as e:
...     logging.exception("Something unexpected happened", e)
... 
--- Logging error ---
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
ZeroDivisionError: division by zero

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.10/logging/__init__.py", line 1100, in emit
    msg = self.format(record)
  File "/usr/lib/python3.10/logging/__init__.py", line 943, in format
    return fmt.format(record)
  File "/usr/lib/python3.10/logging/__init__.py", line 678, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.10/logging/__init__.py", line 368, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "<stdin>", line 4, in <module>
Message: 'Something unexpected happened'
Arguments: (ZeroDivisionError('division by zero'),)
>>> 
```

I fixed the exception logging in both `deye_connector.py` and `deye_daemon.py` now.